test/integration/max-staleness/max_staleness.test.js

```diff
@@ -46,33 +46,31 @@ describe('Max Staleness', function () {
     metadata: {
       requires: {
         generators: true,
-        topology: 'single'
+        topology: 'replicaset'
       }
     },
 
-    test: function (done) {
-      var self = this;
+    test: function () {
+      const self = this;
       const configuration = this.configuration;
       const client = configuration.newClient(
         `mongodb://${test.server.uri()}/test?readPreference=secondary&maxStalenessSeconds=250`,
         { serverApi: null } // TODO(NODE-3807): remove resetting serverApi when the usage of mongodb mock server is removed
       );
 
-      client.connect(function (err, client) {
+      client.connect(async function (err, client) {
```

Comment:
Passing an async function into connect I don't think will propogate any rejections correctly. you could either finish converting the test function to async/await or move this function to a `.then` chain, and make sure you `return` the promise